src/libraries/Common/src/System/Security/Cryptography/RSAAndroid.cs

```diff
@@ -422,47 +426,56 @@ public override void ImportParameters(RSAParameters parameters)
 
                 FreeKey();
                 _key = new Lazy<SafeRsaHandle>(key);
-
-                // Use ForceSet instead of the property setter to ensure that LegalKeySizes doesn't interfere
-                // with the already loaded key.
-                ForceSetKeySize(BitsPerByte * Interop.AndroidCrypto.RsaSize(key));
+                SetKeySizeFromHandle(key);
             }
 
-            public override void ImportRSAPublicKey(ReadOnlySpan<byte> source, out int bytesRead)
+            public override unsafe void ImportRSAPublicKey(ReadOnlySpan<byte> source, out int bytesRead)
             {
                 ThrowIfDisposed();
 
-                int read;
-
-                try
-                {
-                    AsnDecoder.ReadEncodedValue(
-                        source,
-                        AsnEncodingRules.BER,
-                        out _,
-                        out _,
-                        out read);
-                }
-                catch (AsnContentException e)
+                fixed (byte* ptr = &MemoryMarshal.GetReference(source))
                 {
-                    throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding, e);
-                }
-
-                SafeRsaHandle key = Interop.AndroidCrypto.DecodeRsaPublicKey(source.Slice(0, read));
-
-                if (key is null || key.IsInvalid)
-                {
-                    throw new CryptographicException();
-                }
+                    using (MemoryManager<byte> manager = new PointerMemoryManager<byte>(ptr, source.Length))
+                    {
+                        ReadOnlyMemory<byte> subjectPublicKey;
+                        try
+                        {
+                            AsnReader reader = new AsnReader(manager.Memory, AsnEncodingRules.BER);
+                            subjectPublicKey = reader.PeekEncodedValue();
+                        }
+                        catch (AsnContentException e)
+                        {
+                            throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding, e);
+                        }
```

Comment:
All of this (except the catch-normalization) can be replaced with a call to AsnDecoder.TryReadEncodedValue.  (It takes a ReadOnlySpan, instead of a ReadOnlyMemory).

It's just new API, and we didn't go back and clean up places using PointerMemoryManager where they didn't need to anymore.

Reply:
Oh, but we need ReadOnlyMemory for SubjectPublicKeyInfoAsn.SubjectPublicKey.

This is probably the easiest way, then :smile:.